lib/routes/totp.js

```diff
@@ -248,6 +272,37 @@ module.exports = (log, db, customs, config) => {
             request.emitMetricsEvent('totpToken.unverified', {uid: uid})
           }
         }
+
+        function sendEmailNotification() {
+          return db.account(sessionToken.uid)
+            .then((account) => {
+              const geoData = request.app.geo
+              const ip = request.app.clientAddress
+              const service = request.payload.service || request.query.service
+              const emailOptions = {
+                acceptLanguage: request.app.acceptLanguage,
+                ip: ip,
+                location: geoData.location,
+                service: service,
+                timeZone: geoData.timeZone,
+                uaBrowser: request.app.ua.browser,
+                uaBrowserVersion: request.app.ua.browserVersion,
+                uaOS: request.app.ua.os,
+                uaOSVersion: request.app.ua.osVersion,
+                uaDeviceType: request.app.ua.deviceType,
+                uid: sessionToken.uid
+              }
+
+              // Check to see if this token was just verified, if it is, then this means
+              // the user has enabled two step authentication, otherwise send new device
+              // login email.
+              if (isValidCode && ! tokenVerified) {
+                return mailer.sendPostAddTwoStepAuthNotification(account.emails, account, emailOptions)
+              }
+
+              return mailer.sendNewDeviceLoginNotification(account.emails, account, emailOptions)
```

Comment:
Does this mean we'll send this email twice? Once from `/account/login` and then again from here? Do we want to do that?

Reply:
This shouldn't happen because on the original login request, the user would not have technically logged successfully (auth-server returns `verificationMethod`).